File: Data Structures & Algorithms/anagram-groups/submission-2.cs
public class Solution
{
    public List<List<string>> GroupAnagrams(string[] strs)
    {
        Dictionary<string,List<string>> mp = new Dictionary<string, List<string>>();
        foreach (var str in strs)
        {

            char[] chars = str.ToCharArray();
            Array.Sort(chars);
            string temp= new string(chars);

            if (mp.ContainsKey(temp))
            {
                mp[temp].Add(str);
            }
            else mp[temp] = new List<string>() { str};
        }

        List<List<string>> res= new List<List<string>>();
        foreach (var it in mp)
        {
            res.Add(it.Value);
        }
        return res;
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-2.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string res = "";
        for (string str : strs)
        {
            res += str + '.';
        }
        return res;
    }

    vector<string> decode(string s) {
        vector<string> res;
        string temp = "";
        for (char c : s)
        {
            if (c == '.') 
            {
                res.push_back(temp);
                temp = "";
            }
            else
            {
                temp += c;
            }       
        }
        return res;
    }
};
